demo/lego_official_projects/common.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+import logging
+import sys
+import time
+from ev3dev.auto import *
+
+log = logging.getLogger(__name__)
+
+
+class Tank(object):
+
+    def __init__(self, left_motor, right_motor):
+
+        for motor in (left_motor, right_motor):
+            if motor not in OUTPUTS:
+                log.error("%s in an invalid output, choices are %s" % (motor, ''.join(OUTPUTS)))
+                sys.exit(1)
+
+        self.left_motor = LargeMotor(left_motor)
+        self.right_motor = LargeMotor(right_motor)
+
+        for x in (self.left_motor, self.right_motor):
+            if not x.connected:
+                log.error("%s is not connected" % x)
```

Comment:
There is no such thing. The `address` is no different from any other attribute when device is connected. One can ask for a motor on port 'outA', or for a motor that has zero speed at the moment. The [`Device` constructor](https://github.com/rhempel/ev3dev-lang-python/blob/master/ev3dev/core.py#L139) just takes a bunch of keyword arguments, and tries to match each of those to a sysfs attribute.

What we could do, is to save the actual `kwargs` used for device initialization, and then use them inside `__str__` or `__repr__`. So that
```py
"%s" % LargeMotor(address='outA') == "LargeMotor(address=outA)"
```
and
```py
"%s" % LargeMotor(speed=0) == "LargeMotor(speed=0)"
```
@rhempel, what do you think?